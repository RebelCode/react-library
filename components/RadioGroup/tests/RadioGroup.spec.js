import React from "react";
import { render, cleanup } from "@testing-library/react";
import { RadioGroup } from "../index";

afterEach(cleanup);

describe("RadioGroup", () => {
  const Comp = (
    <RadioGroup
      name="multiChoice"
      value="multipleDaily"
      options={[
        {
          value: "multipleDaily",
          label: "Multiple times a day.",
        },
      ]}
    />
  );

  it("should take a snapshot", () => {
    const { asFragment } = render(Comp);

    expect(asFragment(Comp)).toMatchSnapshot();
  });

  it("renders correctly", () => {
    const { queryByLabelText } = render(Comp);

    expect(queryByLabelText("RadioGroup")).toBeTruthy();
  });
});
