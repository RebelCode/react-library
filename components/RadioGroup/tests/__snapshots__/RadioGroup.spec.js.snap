// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup should take a snapshot 1`] = `
<DocumentFragment>
  <div
    aria-label="RadioGroup"
    class="radioGroup "
  >
    <label
      class="option"
    >
      <input
        checked=""
        name="multiChoice"
        type="radio"
        value="multipleDaily"
      />
      <span>
        Multiple times a day.
      </span>
    </label>
  </div>
</DocumentFragment>
`;
